[whitesnowx] iP

whitesnowx opened the pull request:
Name: Add user input feature, task delete, save, mark, unmark, delete
Description: 
# Luna ☾ 
Luna is your trusty friend to make sure you are always on track and do not miss any important activities.

Luna is
* User-friendly
* Cute
* ~~a Virus~~ Safe

*How to get started:*
1. Download from [here](https://github.com/whitesnowx/ip/releases/tag/A-Jar)
2. Double-click the file.
3. Add your tasks
4. Let your _worries_ disappear!

  ∧,,,∧\
(  ̳• · • ̳)\
/    づ♡ \

#### Features
- [x] Add Tasks
- [x] Delete Tasks
- [x] Find Tasks
- [x] List Tasks
- [x] Save/Store List
- [ ] Sort Task (maybe)

This is the `greetingString()` message which constructs the first message you should see when you run the file but sadly some characters might be lost in transmission :"
```
private  String greetingString() {
        String msg = "";
        msg += String.format("%30s", " （„• ֊ •„)♡\n");
        msg += "heyo! my name is ✦" + this.name + "✦\n";
        msg += "What would you like for me to do???\n";
        return msg;
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

whitesnowx committed:
Added greetings, horizontal line and sendoff text

whitesnowx committed:
Added greetings, horizontal line and sendoff text

whitesnowx committed:
Added greetings, horizontal line and sendoff text

whitesnowx committed:
Added greetings, horizontal line and sendoff text

whitesnowx committed:
Added greetings, horizontal line and sendoff text

whitesnowx committed:
Added greetings, horizontal line and sendoff text

whitesnowx committed:
Level 1 echo implemented!

whitesnowx committed:
Level 1 echo implemented!
(reduce some warnings)

whitesnowx committed:
Level 1 echo implemented!
(reduce some warnings)

whitesnowx committed:
Level 1 echo implemented!
(reduce some warnings)

whitesnowx committed:
Level 2 - Completed

No Warnings

whitesnowx committed:
Level 3 - Completed
(mark and unmarking)
No Warnings

whitesnowx committed:
Level 4 - Completed

(different types of tasks)
No Warnings! (typos tho)

whitesnowx committed:
Level 5 - Completed
--> Same as Level 4
--> already have some error handling
--> can come back to stretch goals)
(different types of errors)
No Warnings! (typos tho)

whitesnowx committed:
Level 6 - Completed

(Delete feature)
No Warnings! (typos tho)

whitesnowx committed:
[op] A-Enum - Completed

(Use for providing hints for correct command syntax)

whitesnowx committed:
[op] A-Enum - Completed

(Use for providing hints for correct command syntax)

whitesnowx committed:
A-TextUITesting - Completed

(Use for providing hints for correct command syntax)

whitesnowx committed:
Merge remote-tracking branch 'origin/master'

whitesnowx committed:
A-TextUITesting - Completed

(Use for providing hints for correct command syntax)

whitesnowx committed:
Merge remote-tracking branch 'origin/master'

whitesnowx committed:
Remove some blank lines

whitesnowx committed:
created ListFileManager

Test read/write/create

whitesnowx committed:
create ListFileManager

Add createFile() which is called during construction to create the txt file at data
Add appendToFile() which appends a given string to the file
Add appendEntry which formats an entry to a string and is appended to the file (WIP)

whitesnowx committed:
update Load function

- clear the current arraylist and replaces from list

save function
- require changing append to write (clears the previous data)

whitesnowx committed:
rename main project from the default Duke to Luna

whitesnowx committed:
Clean up codes
- remove temporary codes

add ListFileManager.clearFile() function
- clears the current file in the hard drive
- facilitate save feature with appendEntry feature

whitesnowx committed:
Clean up codes
- remove temporary codes

add ListFileManager.clearFile() function
- clears the current file in the hard drive
- facilitate save feature with appendEntry feature

ThatGirlJam commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+        while (!exitFlag) {
+            String input_command = br.readLine();
+            if (input_command.equalsIgnoreCase("help")) {
+                switch(currentHint) {
```
Hi! Perhaps could configure IDE to align "switch" and "case" on the same indentation, according to the code standard?

ThatGirlJam commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+            return msg;
+        }
+
+        public static String sad_cat_img() {
```
I like the very cute cat and very apt name 🐱 🐈 😿

ThatGirlJam commented on src/main/java/list_Entry.java:
```diff
@@ -0,0 +1,42 @@
+public class list_Entry {
```
May have to change the class name to fit Java class naming standards? Eg, ListEntry

ThatGirlJam commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+            return msg;
+        }
+
+        public static void shifted_print(String text) throws Exception{
```
May have to change naming to fit function name standards in Java? 

```suggestion
public static void shiftPrint(String text) throws Exception { 
```

jessicawyz replied:
you can also consider adding javaDoc here since shiftPrint is not a very clear indication of what this function is doing

ThatGirlJam commented on src/main/java/list_Entry_Todo.java:
```diff
@@ -0,0 +1,11 @@
+public class list_Entry_Todo extends list_Entry {
+
+    public list_Entry_Todo(String task, boolean check) {
```
May be good to add more JavaDocs to explain the organisation and purpose of the different methods/classes?

ThatGirlJam commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    enum commandHints {
```
I like the use of Enums! 👍

ThatGirlJam reviewed (commented):
I like the very cute idea and personality of the bot! Some changes could be made to be even more aligned with the coding standard / Java standards 😄 Good job so far!

jessicawyz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+    }
+
+
+    public static class bot_functions {
```
perhaps use PascalCase in class naming

iynixil replied:
I like the separation of the bot functions, makes it easier to extract this out when separating into more OOP classes later on

jessicawyz commented on src/main/java/list_Entry.java:
```diff
@@ -0,0 +1,42 @@
+
+    public list_Entry() {
+    }
+    public list_Entry(String task, boolean check, String type, String tStart, String tEnd) {
```
yea maybe change the class name so it is PascalCase

jessicawyz commented on src/main/java/list_Entry_Deadline.java:
```diff
@@ -0,0 +1,10 @@
+public class list_Entry_Deadline extends list_Entry {
```
hi maybe can change the class naming here too :>

jessicawyz reviewed (commented):
Hi I love the cute cat idea haha, maybe you can edit the class names to better follow java standards and also add more javaDoc explaining what the methods do :))

iynixil commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+            return msg;
+        }
+
+        public static void shifted_print(String text) throws Exception{
```
Seems like this is a modified printing function for showing feedback to user, like that this is separated as it makes printing easier for various commands, good job!

iynixil commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+        MARK,
+        EXIT
+    }
+    public static void main(String[] args) throws Exception{
```
May have forgotten a space for `Exception{` -> `Exception {`, noticed this in a few other methods too

iynixil reviewed (commented):
LGTM. Nice separation of some methods, and a few nits to fix.

whitesnowx committed:
Test punch

Ryan-Reno commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+        ArrayList<list_Entry> user_list = new ArrayList<>();
+
+        while (!exitFlag) {
+            String input_command = br.readLine();
```
The string naming should be in camelCase. So in this case it should be inputCommand instead. I think underscore is mainly used for constants

Ryan-Reno commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+        String NAME = "Luna"; // TENTATIVE
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        bot_functions.shifted_print(bot_functions.greetingString(NAME));
+        boolean exitFlag = false;
```
Boolean names should sound like a boolean. So you could instead change it to isExit, to indicate if it is an Exit input or not

Ryan-Reno commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,220 @@
+                        bot_functions.shifted_print("To add a deadline task to your list\n--> Type: deadline <task name> <end date>");
+                        break;
+                    case EVENT:
+                        bot_functions.shifted_print("To add a event task to your list\n--> Type: event <task name> <start date> <end date>");
```
Although the hard limit for characters in a line is 120, the number of characters for this line is 117. Try to break it up into two lines for better readability and try to avoid such long code within a single line

Ryan-Reno reviewed (commented):
Nice job in doing your iP! But try and make sure to follow naming conventions that have been stated within the Java Coding standard!

whitesnowx committed:
Update String taskStart taskEnd to be LocalDate

whitesnowx committed:
Merge branch 'branch-Level-8'

whitesnowx committed:
Merge branch 'branch-Level-8'

whitesnowx committed:
Merge remote-tracking branch 'origin/master'

whitesnowx committed:
Extract botfunctions class from main to new class Ui

whitesnowx committed:
Refractor ListFileManager to Storage

whitesnowx committed:
Implement Parser, Ui, TaskList, Command
(Missing Save/Delete/Load Commands and Storage)

whitesnowx committed:
Implement DeleteCommand

whitesnowx committed:
Implement SaveCommand, LoadCommand, Storage

whitesnowx committed:
Seperate Run and start which run once

whitesnowx committed:
Move all java files in Folder called Luna and package them as Luna

whitesnowx committed:
Merge branch 'add-gradle-support'

whitesnowx committed:
(I will be back)

whitesnowx committed:
add JUnit 2 Test for ParserTest and TaskListTest each

whitesnowx committed:
add JUnit 2 Test for ParserTest and TaskListTest each

whitesnowx committed:
Add JavaDocs to all the Commands File and Tasklist, UI, ListEntry, Parser, Storages

whitesnowx committed:
Remove dead space, whitespace and old comments

whitesnowx committed:
Add JavaDocs to all the Commands File and Tasklist, UI, ListEntry, Parser, Storages

whitesnowx committed:
Fix warnings and project problems

whitesnowx committed:
Fix warnings and project problems

whitesnowx committed:
Add some missing periods in javadocs

whitesnowx committed:
Change check to isDone to match coding standards

whitesnowx committed:
Implement Find

whitesnowx committed:
Merge branch 'branch-A-JavaDoc'

whitesnowx committed:
Merge branch 'branch-A-CodingStandard'

whitesnowx committed:
Implement Find

whitesnowx committed:
Refactor Cat

whitesnowx committed:
Reformat Cat

whitesnowx committed:
Update README.md

Change from the Template to demonstrate some GFMD

whitesnowx committed:
Update README.md

Shift safe up to beside a virus

whitesnowx committed:
Update README.md

Update download here to have hyperlink to the jar generation

whitesnowx committed:
Fix formatting for spaces

whitesnowx committed:
Change UI to be compatible with GUI

whitesnowx committed:
Update README.md

Add missing inline code block

whitesnowx committed:
Remove old commented codes

Clean codes is better for readability

Decrease the number of unused lines

Lets
* keep only necessary comments

whitesnowx committed:
Refract image files

Easy for differentiate purpose

image file for user and luna is renamed  to their respective name

whitesnowx committed:
Add Assertions

Lack of Assertions is not safe.

Increasing number of Assertions is safer.

Lets
* add more assertions

Added to Storage.java, TaskList.java, Luna.java

whitesnowx committed:
Remove Whitespace and dead codes

Code Quality increase readability

Check all flies for dead codes and extra lines

whitesnowx committed:
Fix warnings by debugger and format imports

Organisation of import to understand dependencies. Fix warnings to reduce redundant and maximise code efficiency.

whitesnowx committed:
Fix more warnings and refactor getter methods

Provide clarity to class methods and getter functions. Reduce redundant code logic.

whitesnowx committed:
Merge pull request #2 from whitesnowx/branch-A-CodeQuality

Branch a code quality

whitesnowx committed:
Merge pull request #3 from whitesnowx/branch-A-Assertions

Add Assertions

whitesnowx committed:
Add snoozeEntry function to ListEntry, ListEntryDeadline, ListEntryEvent

Allocate methods for the entry to snooze itself.

Prepare for luna to access the entry to be snoozed.

whitesnowx committed:
Implement snooze function.

Allow user to use the snooze function through the use of SnoozeCommand and added snooze methods.

Prevent user from snoozing entries without an end date.

Prevent user from using the prompting the snooze command without the needed inputs.

whitesnowx committed:
Abide photo regulations

Do not use free online images unless permission to.

Change both user and luna image files. File type is now both jpg, changes are reflected in the code.

whitesnowx committed:
Add ui outputs to commands

Some successfully executed command did not have any output.

Include information on the type of command that was just executed by the user.

whitesnowx committed:
Update README.md

Outline of User Guide

whitesnowx committed:
Update README.md

Update the basic feature description

whitesnowx committed:
Update README.md

Update Quick Start and Feature Summary

whitesnowx committed:
Update README.md

Remove personal todo list

whitesnowx committed:
Add files via upload

whitesnowx committed:
Rename ui.png to Ui.png

whitesnowx committed:
Update README.md

whitesnowx committed:
Add files via upload

whitesnowx committed:
Add files via upload

whitesnowx committed:
Update README.md

whitesnowx committed:
Update README.md

whitesnowx committed:
Fix Checkstyle

whitesnowx committed:
Fix warnings

whitesnowx committed:
Format into folder

Shift relevant child files into Command folder and child listEntry into entry folders

whitesnowx committed:
Fix grandlew and jar generation
